### PR TITLE
[AIR] Add AIR install extra

### LINKED ISSUE #23439: Should add a ray[air] package that installs everything.
`pip install ray[air]` instead of `pip install ray[train, etc...]`

### PR DESCRIPTION
**Pull request by amogkam**
Closes https://github.com/ray-project/ray/issues/23439

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -255,6 +255,11 @@ def get_packages(self):
         "scipy",
     ]
 
+    # Ray AI Runtime should encompass Data, Tune, and Serve.
+    setup_spec.extras["air"] = (
+        setup_spec.extras["tune"] + setup_spec["data"] + setup_spec.extras["serve"]
```
Comment (richardliaw) [inline python/setup.py:260]:
```suggestion
        setup_spec.extras["tune"] + setup_spec.extras["data"] + setup_spec.extras["serve"]
```

?

Comment (amogkam) [reply python/setup.py:260]: Oops, yep

**Review by richardliaw — APPROVED** [review]

**Review by ericl — APPROVED** [review]

File: python/setup.py [hunk 2]
```diff
@@ -255,6 +255,11 @@ def get_packages(self):
         "scipy",
     ]
 
+    # Ray AI Runtime should encompass Data, Tune, and Serve.
+    setup_spec.extras["air"] = (
+        setup_spec.extras["tune"] + setup_spec.extras["data"] + setup_spec.extras["serve"]
+    )
```
Comment (simon-mo) [inline python/setup.py:261]: I would recommend doing the list(set(... )) trick to deduplicate (see below). Tune and Serve have overlapping dependencies.

**Comment by amogkam** [comment]
@simon-mo that was intentional-- we should make sure that we don't have conflicting dependencies across the ray libraries right?

**Comment by simon-mo** [comment]
oh neither library had it pinned. it's just both depends on `requests` at the moment and I don't know whether setup_spec extras allow duplicated entries in the list.

**Comment by amogkam** [comment]
oh you're right list(set(...)) will still fail even if there's conflicting dependencies. Sounds good, will make the change!

**Pull request merged**